components/src/core/components/TableFilter/TableFilter.vue

```diff
@@ -44,24 +46,70 @@ export default defineComponent({
       type: String,
       required: true,
     },
+    itemsSelected: {
+      type: Boolean,
+      default: false,
+    },
   },
 
-  setup() {
+  setup(props) {
     const responsiveState = useResponsive();
     const isActive = ref(true);
+    const isSpilled = ref(true);
 
     const toggleFilters = () => {
       isActive.value = !isActive.value;
     };
 
+    const updateIsSpilled = () => {
+      nextTick(function() {
+        const titleWidth = document.getElementsByClassName(
+          'oxd-table-filter-header-title',
+        )[0]
+          ? document.getElementsByClassName('oxd-table-filter-header-title')[0]
+              .clientWidth
+          : 0;
+        const actionButtonWidth = document.getElementsByClassName(
+          'oxd-table-filter-header-bulk-actions',
+        )[0]
+          ? document.getElementsByClassName(
+              'oxd-table-filter-header-bulk-actions',
+            )[0].clientWidth
+          : 0;
+        const fullWidth = document.getElementsByClassName(
```

Comment:
Isn't document.getElementsByClassName is a costly operation to have two times in a ternary operation? @Super-Chama any thoughts? Do we need to assign it to a variable and reuse it?

Reply:
I think the browser is actively traversing all nodes to find it so it might by O(n) kind of operation. but in here if these nodes are conditionally rendered i don't think we can assign to a variable and use. if not its possible to cache it in function scope with variables.

Reply:
Yeah it is conditional. @lgtkaushalya Yes, I will add it to a single call. by adding it to a variable.